src/data_info/clients/async_tasks.clj

```diff
@@ -34,17 +35,28 @@
   [filters]
   (async-tasks-client/get-by-filter (config/async-tasks-client) filters))
 
+(defn run-async-thread
+  [async-task-id thread-function prefix]
+  (let [^Runnable task-thread (fn [] (thread-function async-task-id))]
+    (.start (Thread. task-thread (str prefix "-" (string/replace async-task-id #".*/tasks/" "")))))
+  async-task-id)
+
 (defn paths-async-thread
-  [async-task-id jargon-fn]
-  (let [{:keys [username] :as async-task} (get-by-id async-task-id)
-        update-fn (fn [path action]
-                    (log/info "Updating async task:" async-task-id ":" path action)
-                    (add-status async-task-id {:status "running" :detail (format "%s: %s" path (name action))}))]
-    (try+
-      (add-status async-task-id {:status "started"})
-      (irods/with-jargon-exceptions :client-user username [cm]
-        (jargon-fn cm async-task update-fn))
-      (add-completed-status async-task-id {:status "completed"})
-      (catch Object _
-        (log/error (:throwable &throw-context) "failed processing async task" async-task-id)
-        (add-completed-status async-task-id {:status "failed"})))))
+  ([async-task-id jargon-fn]
+   (paths-async-thread async-task-id jargon-fn true))
+  ([async-task-id jargon-fn use-client-user?]
```

Comment:
Adding this extra argument means we can do async tasks without a client user set, which is needed for deletes.